packages/react/react/src/hooks/useLazyImage.ts

```diff
@@ -0,0 +1,111 @@
+import { noop } from '@toss/utils';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { usePreservedCallback } from './usePreservedCallback';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onLoadComplete?: () => void;
+  onInView?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useLazyImage({
+  src,
+  rootMargin,
+  threshold,
+  root,
+  onLoadComplete: _onLoadComplete,
+  onInView: _onInView,
+}: Props) {
+  const ref = useRef<HTMLImageElement>(null);
+
+  const onInView = usePreservedCallback(_onInView ?? noop);
+  const onLoadComplete = usePreservedCallback(_onLoadComplete ?? noop);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const onLoadImage = useCallback(
+    (element: HTMLImageElement) => {
+      setIsLoading(true);
+
+      element.onload = () => {
+        setIsLoading(false);
+        onLoadComplete();
+      };
+    },
+    [onLoadComplete]
+  );
```

Comment:
I think this is closer to 'register' an event handler, than to be called when an event fires. So I think `registerLoadEventHandler` is more correct name.

Reply:
@HoseungJang 
Oh... That's right

I think your suggestion is more appropriate since it involves registering an "onload" event handler on the img element.

I'll fix that code.